app/translators/base.translator.js

```diff
@@ -1,20 +1,35 @@
 'use strict'
 
+const Boom = require('@hapi/boom')
+
 class BaseTranslator {
   constructor (data) {
-    Object.assign(this, { _data: data })
-    // Set _data to be non-enumerable so it isn't passed to model validation
+    // Validate data against this._schema and throw bad data error if validation fails
+    const { error, value: validatedData } = this._validate(data)
+    if (error) {
+      throw Boom.badData(error)
+    }
+
+    // Assign validated data to _data and set it to be non-enumerable so it isn't visible outside of the translator
+    Object.assign(this, { _data: validatedData })
     Object.defineProperty(this, '_data', { enumerable: false })
 
+    // Create each getter in _translations. We do this within the constructor to keep Joi happy
+    // TODO: We may not need to do this here now that validation is done on the presenter
```

Comment:
Agh, I keep saying presenter instead of translator! The TODO is accurate with that in mind -- I originally placed the creation of getters within the constructor as creating them outside of it caused validation to fail when passing the presenter to the model. So in theory we could now move creation of the getters outside of the constructor now that validation is done by the translator. However I don't recall right now if that's even necessary/best practice to do so.